src/pages/StocktakeEditPage.js

```diff
@@ -422,8 +422,26 @@ export class StocktakeEditPage extends React.Component {
     this.setState({ stocktakeItem: item, isStocktakeEditModalOpen: true });
   };
 
+  renderReasonModal = () => {
+    const { currentStocktakeItem, isReasonsModalOpen, reasons } = this.state;
+    const currentReasonIndex = reasons.findIndex(
```

Comment:
I'm wondering - I have added a `highlightValue` property to `GenericChooseModal` (as well as a couple of other things) in the vaccines branch. Wondering if it is better for me to make the changes to this component sooner rather than later? Not a huge difference here, but is something

Reply:
Wouldn't rush, though if the API is changing then will have to test reason modal with vaccine PR for sure!

Reply:
Last thing is just the `highlightIndex` - I think it just has a nicer look if the `highlightIndex` is set to `0` when there is no reason selected. Couldn't think of a cool JS way :( `highlightIndex={currentReasonIndex >= 0 ? currentReasonIndex : 0}`. 

This defaulting to 0 is not part of the `GenericChooseModal` (Neither this version nor the vaccine version....) as I thought being 'generic' it should have as little behaviour pre-defined as possible although that line of thinking might be wrong!

i.e. this view when no reasons have been applied yet
![image](https://user-images.githubusercontent.com/35858975/58588620-29e35800-82b4-11e9-92e0-e0a318db44b4.png)

Over this view (what it is with this changes)
![image](https://user-images.githubusercontent.com/35858975/58588846-b42bbc00-82b4-11e9-806b-d5fe1fdbafef.png)

I can see how this could *potentially* be confusing, but I think it's convention at this point for a drop down to have the first item selected

Reply:
Currently with my changes I get neither of those pictures. There is just no highlighting or shadow at all.

![image](https://user-images.githubusercontent.com/7684221/58598791-d8e25c80-82d1-11e9-92cc-baaae7220340.png)

Reply:
maybe I did something to make the second pic highlight come up - I am suggesting that the first highlight should be used